app/helpers/profile_helper.rb

```diff
@@ -0,0 +1,16 @@
+module ProfileHelper
+  def social_authentication_links_for(user)
+    # Returns a Hash containing URLs for social authentication providers
+    # Currently supported: GitHub, Twitter, Facebook
+    user_identities = user.identities_enabled
+    return {} if user_identities.blank?
+
+    urls = user_identities.pluck(:auth_data_dump).each_with_object({}) do |data, hash|
+      if (data_urls = data.dig(:info, :urls))
+        hash.merge!(data_urls)
+      end
+    end
+
+    { github: urls["GitHub"], twitter: urls["Twitter"], facebook: urls["Facebook"] }.compact
```

Comment:
note: at some point - if we need to add more than these three social links, it makes sense to extract this mapping of permitted response keys to the user identity url's keys  into a configuration or constant. Nothing to do just now, just pointing out this _could_ be a place to refactor if we're editing this frequently.